pulsar-broker/src/test/java/org/apache/pulsar/broker/service/persistent/ShadowTopicTest.java

```diff
@@ -0,0 +1,92 @@
+    @Override
+    protected void setup() throws Exception {
+        baseSetup();
+    }
+
+    @AfterClass(alwaysRun = true)
+    @Override
+    protected void cleanup() throws Exception {
+        internalCleanup();
+    }
+
+    @Test()
+    public void testNonPartitionedShadowTopicSetup() throws Exception {
+        String sourceTopic = "persistent://prop/ns-abc/source";
+        String shadowTopic = "persistent://prop/ns-abc/shadow";
+        //1. test shadow topic setting in topic creation.
+        admin.topics().createNonPartitionedTopic(sourceTopic);
+        admin.topics().createShadowTopic(shadowTopic, sourceTopic);
+        PersistentTopic brokerShadowTopic =
+                (PersistentTopic) pulsar.getBrokerService().getTopicIfExists(shadowTopic).get().get();
+        Assert.assertEquals(brokerShadowTopic.getShadowSourceTopic().toString(), sourceTopic);
+        Assert.assertEquals(admin.topics().getShadowSource(shadowTopic), sourceTopic);
+
+        //2. test shadow topic could be properly loaded after unload.
+        admin.namespaces().unload("prop/ns-abc");
+        Assert.assertTrue(pulsar.getBrokerService().getTopicReference(shadowTopic).isEmpty());
+        Assert.assertEquals(admin.topics().getShadowSource(shadowTopic), sourceTopic);
+        brokerShadowTopic = (PersistentTopic) pulsar.getBrokerService().getTopicIfExists(shadowTopic).get().get();
+        Assert.assertEquals(brokerShadowTopic.getShadowSourceTopic().toString(), sourceTopic);
+    }
+
+    @Test()
+    public void testPartitionedShadowTopicSetup() throws Exception {
+        String sourceTopic = "persistent://prop/ns-abc/source-p";
+        String shadowTopic = "persistent://prop/ns-abc/shadow-p";
+        String shadowTopicPartition = TopicName.get(shadowTopic).getPartition(0).toString();
+
+        //1. test shadow topic setting in topic creation.
+        admin.topics().createPartitionedTopic(sourceTopic, 2);
+        admin.topics().createShadowTopic(shadowTopic, sourceTopic);
+        pulsarClient.newProducer().topic(shadowTopic).create().close();//trigger loading partitions.
+        PersistentTopic brokerShadowTopic = (PersistentTopic) pulsar.getBrokerService()
+                .getTopicIfExists(shadowTopicPartition).get().get();
+        Assert.assertEquals(brokerShadowTopic.getShadowSourceTopic().toString(), sourceTopic);
+        Assert.assertEquals(admin.topics().getShadowSource(shadowTopic), sourceTopic);
+
+        //2. test shadow topic could be properly loaded after unload.
+        admin.namespaces().unload("prop/ns-abc");
+        Assert.assertTrue(pulsar.getBrokerService().getTopicReference(shadowTopic).isEmpty());
+
+        Assert.assertEquals(admin.topics().getShadowSource(shadowTopic), sourceTopic);
+        brokerShadowTopic =
+                (PersistentTopic) pulsar.getBrokerService().getTopicIfExists(shadowTopicPartition).get().get();
+        Assert.assertEquals(brokerShadowTopic.getShadowSourceTopic().toString(), sourceTopic);
```

Comment:
It's better verify the `ManagedLedger `of `brokerShadowTopic` is a `ShadowManagedLedgerImpl`